build.sbt

```diff
@@ -164,7 +164,7 @@ lazy val scalaProbePlugin =
     .dependsOn(probePlugin, scalaProbeApi)
```

Comment:
Thanks for implementing this. It looks good overall, I left some comments.

First thing to know is that ide-probe can be configured from config and from code (just create IntelliJFixture from code). IJ fixture can be also tweaked from code after config is loaded with some helper methods. The biggest bug I see is that if someone modifies list of plugins in IJ fixture it won't be reflected, because it is only read from config. Similarly IJ version can't be changed, it is now part of factory. For existing IJ this can't be specified.
I think high level api should be
```scala
fixture.version
IJFixture(intelliJProvider = IntelliJProvider(IntelliJVersion(..., ...)), ...) // apply method that creates default provider, existing workspace is a rare case
IJFixture(intelliJProvider = IntelliJProvider.Existing(path = ..., ...)) // create existing workspace specifying path
fixture.withVersion(IJVersion(..., ...)) // repaces version in default provider if it is default or throws for Existing
```
This is how I imagine code API, config API is already fine.

Also I am concerned about the design of IntelliJ provider and how it contains the factory in both cases.

This looks like a design problem. IntelliJFactory is meant to create InstalledIJ instance that has everything needed to launch it. It was designed to install isolated instance and this is what most of its methods do. Making it a member of existing workspace, that in theory shouldn't need any of these methods and making it a public api (that IJProvider has Factory) is not ideal. 
What I would rather do (I explain the path as if your PR did not exist, you may want to just alter your code to not start from scratch to get to the same result) is:
- Rename IJFactory to IJProvider fo consistency with WorkspaceProvider.
- Split into interface and implementation extracting public api that makes sense for both future implementations to interface. 
-Move whatever needs to be moved to another place, for example version shouldn't be a field of fixture bit be an accessor method and a setter should throw if it can't replace it. Any such changes that make sense.
- Implement the interface for IJProvider.Existing handling anything that needs to be handled.
- If both interface need some common logic, extract it to a separate class, named approprietly after what it is doing. For example PluginInstaller with a clear API.

Also avoid making private methods public just to reuse them or for testing, if these methods are used by other public methods. For example as you did with `version` resolving from file. It would be better to create a separate class, like IJVersionResolver that given a root path of intellij installation can return a version, and test it in an unit test.

Reply:
Will be done in the next commit.

Reply:
Done.